portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -37,23 +37,44 @@
 import java.text.*;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/delete-all-data")
-public class DeleteAllDataServlet extends HttpServlet {
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    UserService userService = UserServiceFactory.getUserService();
-    String isAdmin = String.valueOf(userService.isUserAdmin());
-    if (isAdmin.equals("true")){
-      Query query = new Query("Comment").setKeysOnly();
+    String deleteAll = (String) request.getParameter("deleteAll");
+    if (deleteAll.equals("true")) {
+      UserService userService = UserServiceFactory.getUserService();
+      String isAdmin = String.valueOf(userService.isUserAdmin());
+      if (isAdmin.equals("true")){
+        Query query = new Query("Comment").setKeysOnly();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);  
+        List<Key> keys = new ArrayList<>();  
+        for (Entity entity : results.asIterable()) {
+        keys.add(entity.getKey());
+        }
+        datastore.delete(keys);
+        System.out.println("Deleted all comments");
```

Comment:
Good optimization from the last version.  One more optimization to think of for next time:  you have a few pieces that are the same here and on the lines 62-75 below.  Mainly checking for the admin flag, and the actual delete action.  The only difference is the key you pass into the delete:  one passes all keys while the other passes a single key.  It would be good if you can combine similarities into reusable utility methods.  This way if your logic changes (maybe you create a feature for a non-admin user to delete all comments they own), you'd only have to change in one spot.